<a href="https://colab.research.google.com/github/dorothyomare/AUTOLIB-ANALYSIS/blob/master/eldohub_challenge.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


In [18]:
url = 'https://raw.githubusercontent.com/onaio/ona-tech/master/data/water_points.json'
data = pd.read_json(url)
df = pd.DataFrame(data)
df

water_pay                 respondent  ...     simserial  subscriberid
0          no                  community  ...           NaN           NaN
1          no                  community  ...           NaN           NaN
2          no                  community  ...           NaN           NaN
3          no                  community  ...           NaN           NaN
4          no                  community  ...           NaN           NaN
..        ...                        ...  ...           ...           ...
707        no                  community  ...           NaN           NaN
708        no  local_authority community  ...           NaN           NaN
709        no  local_authority community  ...           NaN           NaN
710        no                  community  ...  8.923301e+18  6.200101e+14
711        no                  community  ...           NaN           NaN

[712 rows x 48 columns]

In [19]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 712 entries, 0 to 711
Data columns (total 48 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   water_pay                    712 non-null    object        
 1   respondent                   712 non-null    object        
 2   research_asst_name           712 non-null    object        
 3   water_used_season            712 non-null    object        
 4   _bamboo_dataset_id           712 non-null    object        
 5   _deleted_at                  0 non-null      datetime64[ns]
 6   water_point_condition        712 non-null    object        
 7   _xform_id_string             712 non-null    object        
 8   other_point_1km              712 non-null    object        
 9   _attachments                 712 non-null    object        
 10  communities_villages         712 non-null    object        
 11  end                          712 non-null    

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Columns: 33669 entries, [{"water_pay" to  "yes"}]
dtypes: object(33669)
memory usage: 0.0+ bytes


In [21]:
data.isnull().sum()

water_pay                        0
respondent                       0
research_asst_name               0
water_used_season                0
_bamboo_dataset_id               0
_deleted_at                    712
water_point_condition            0
_xform_id_string                 0
other_point_1km                  0
_attachments                     0
communities_villages             0
end                              0
animal_number                  184
water_point_id                   0
start                            0
water_connected                  0
water_manager_name               4
_status                          0
enum_id_1                        0
water_lift_mechanism             0
districts_divisions              0
_uuid                            0
grid                             0
date                             0
formhub/uuid                     0
road_available                   0
water_functioning                0
_submission_time                 0
signal              

In [25]:
data.dropna


<bound method DataFrame.dropna of     water_pay                 respondent  ...     simserial  subscriberid
0          no                  community  ...           NaN           NaN
1          no                  community  ...           NaN           NaN
2          no                  community  ...           NaN           NaN
3          no                  community  ...           NaN           NaN
4          no                  community  ...           NaN           NaN
..        ...                        ...  ...           ...           ...
707        no                  community  ...           NaN           NaN
708        no  local_authority community  ...           NaN           NaN
709        no  local_authority community  ...           NaN           NaN
710        no                  community  ...  8.923301e+18  6.200101e+14
711        no                  community  ...           NaN           NaN

[712 rows x 48 columns]>

In [34]:
#number of functional waterpoints
df = data.loc[:, 'water_point_condition':'communities_villages']
df
functional_waterpoints = df.groupby('water_point_condition')
functional_waterpoints.count()

_xform_id_string  ...  communities_villages
water_point_condition                    ...                      
abandoned                            36  ...                    36
broken                               40  ...                    40
functioning                         590  ...                   590
na_dn                                 2  ...                     2
newly_constructed                    28  ...                    28
under_construction                   16  ...                    16

[6 rows x 4 columns]

from the output above, the number of functional waterpoints is 590

In [41]:
#number of waterpoints per community
water_point = df.groupby('communities_villages').water_point_condition.count()
water_point

communities_villages
Abanyeri        4
Akpari-yeri     3
Alavanyo        3
Arigu          12
Badomsa        27
               ..
Zogsa           6
Zua            28
Zuedema        18
Zukpeni         6
Zundem         30
Name: water_point_condition, Length: 65, dtype: int64

In [54]:
#rank of each community by % of broken waterpoints
#no_of_broken_waterpoints = 40   #from the data above
df.sort_values("water_point_condition", inplace = True) 
filter = df["water_point_condition"]=="broken"
df.where(filter, inplace = True) 
df.count


<bound method DataFrame.count of     water_point_condition  ... communities_villages
264                broken  ...                  Soo
403                broken  ...                Guuta
660                broken  ...                Luisa
159                broken  ...            Loagri_1_
130                broken  ...              Tantala
..                    ...  ...                  ...
431                   NaN  ...                  NaN
476                   NaN  ...                  NaN
633                   NaN  ...                  NaN
166                   NaN  ...                  NaN
235                   NaN  ...                  NaN

[712 rows x 5 columns]>

In [42]:
df.groupby('communities_villages').water_point_condition.value_counts()

communities_villages  water_point_condition
Abanyeri              functioning               3
                      newly_constructed         1
Akpari-yeri           functioning               3
Alavanyo              functioning               2
                      abandoned                 1
                                               ..
Zukpeni               abandoned                 3
                      functioning               2
                      broken                    1
Zundem                functioning              29
                      under_construction        1
Name: water_point_condition, Length: 134, dtype: int64